keywords/couchbaseserver.py

```diff
@@ -741,11 +741,14 @@ def recover(self, server_to_recover):
         count = 0
         max_retries = 5
         while count < max_retries:
-            resp = self._session.post(
-                "{}/controller/setRecoveryType".format(self.url),
-                headers={"Content-Type": "application/x-www-form-urlencoded"},
-                data=data
-            )
+            try:
+                resp = self._session.post(
+                    "{}/controller/setRecoveryType".format(self.url),
+                    headers={"Content-Type": "application/x-www-form-urlencoded"},
+                    data=data
+                )
+            except HTTPError:
+                log_info("Got http error while trying to recover the server, so trying one more time")
```

Comment:
you do not want to raise the exception and fail the test ?

Reply:
what's the difference between HTTPError and resp.status_code not equal to 200? I agree with the original code, should not raise exception here because retries are expected.

Reply:
@mgcouchbase .No we should not raise the exception. The idea is to retry . We get http error  which is expected and have retry until it is successful

@euniceatcouchbase . The above also does the same thing what you have mentioned. If http error raised, it will continue and count get increase and it is not status code == 200, so it won't break the loop. Wondering if there is any difference

Reply:
got it, I am fine with the implementation. thumb up.